Add Redshift behavior change: restrict_direct_pg_catalog_access (#6082)
fishtown-analytics/docs.getdbt.com: website/docs/reference/global-configs/behavior-changes.md

  - '{{ ... if flags.WHICH != 'freshness' }}'
```
</File>

## Adapter-specific behavior changes

> 

Reviewer: ```suggestion
Some adapters may show behavior changes when certain flags are enabled. Refer to the following sections for each respective adapter.
```

Reviewer: adding small intro so it's not h2 into h3

### [Redshift] restrict_direct_pg_catalog_access

> Originally, the `dbt-redshift` adapter was built on top of the `dbt-postgres` adapter and used Postgres tables for metadata access. With this flag enabled, the adapter will use the Redshift API (through the Python client) if available, or query Redshift's `information_schema` tables instead.

Reviewer: ```suggestion
Originally, the `dbt-redshift` adapter was built on top of the `dbt-postgres` adapter and used Postgres tables for metadata access. With this flag enabled, the adapter will use the Redshift API (through the Python client) if available, or query Redshift's `information_schema` tables instead of using `pg_` tables.
```

> Originally, the `dbt-redshift` adapter was built on top of the `dbt-postgres` adapter. It leveraged several Postgres tables for accessing metadata. When this flag is enabled, instead of using `pg_` tables, the adapter will use either the Redshift API (via the Python client) if available, and otherwise query tables in Redshift's `information_schema`.

Reviewer: ```suggestion
Originally, the `dbt-redshift` adapter was built on top of the `dbt-postgres` adapter and used Postgres tables for metadata access. With this flag enabled, the adapter will use the Redshift API (through the Python client) if available, or query Redshift's `information_schema` tables instead.
```

> While we don't intend for there to be any user-noticeable behavior changes as a result of this change, out of an abundance of caution, we are gating it behind a behavior-change flag and encouraging users to test it out before it becomes the default for everyone.

Reviewer: ```suggestion
While we don't expect any user-noticeable behavior changes due to this change, out of caution we are gating it behind a behavior-change flag and encouraging users to test it before it becomes the default for everyone.
```
